addons/binding/org.openhab.binding.meteostick/src/main/java/org/openhab/binding/meteostick/handler/meteostickBridgeHandler.java

```diff
@@ -0,0 +1,341 @@
+                            case "#":
+                                break;
+                            case "?":
+                                // Device has been reset - reconfigure
+                                sendToMeteostick(meteostickFormat);
+                                sendToMeteostick(meteostickMode);
+                                sendToMeteostick(meteostickChannels);
+                                break;
+                            default:
+                                if (p.length < 3) {
+                                    logger.debug("MeteoStick bridge: short data ({})", p.length);
+                                    break;
+                                }
+
+                                try {
+                                    meteostickEventListener listener = eventListeners.get(Integer.parseInt(p[1]));
+                                    if (listener != null) {
+                                        listener.onDataReceived(p);
+                                    } else {
+                                        logger.debug("MeteoStick bridge: data from channel {} with no handler",
+                                                Integer.parseInt(p[1]));
+                                    }
+                                } catch (NumberFormatException e) {
+                                }
+                                break;
+                        }
+
+                        updateStatus(ThingStatus.ONLINE);
+
+                        rxCnt = 0;
+                    } else if (rxByte != 10) {
+                        // Ignore line feed
+                        rxPacket[rxCnt] = (byte) rxByte;
+
+                        if (rxCnt < rxPacket.length) {
+                            rxCnt++;
+                        }
+                    }
+                } catch (Exception e) {
+                    rxCnt = 0;
+                    logger.error("Exception during MeteoStick receive thread", e);
+                }
+            }
+
+            logger.debug("Stopping MeteoStick Recieve Thread");
+            serialPort.removeEventListener();
+        }
+    }
+
+    private class OfflineTimerTask extends TimerTask {
+        @Override
+        public void run() {
+            String detail;
+            if (lastData == null) {
+                detail = "No data received";
+            } else {
+                detail = "No data received since " + lastData.toString();
+            }
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, detail);
+        }
+    }
+
+    private synchronized void startTimeoutCheck() {
+        // Stop any existing timer
+        if (offlineTimerTask != null) {
+            offlineTimerTask.cancel();
+            offlineTimerTask = null;
+        }
+
+        // Create the timer task
+        offlineTimerTask = new OfflineTimerTask();
+
+        // Start the timer
+        offlineTimer.schedule(offlineTimerTask, 75000);
```

Comment:
could you use the existing "scheduler" in order to reuse the existing named thread pools?